src/plugins/controls/public/controls/timeslider_control/get_timeslider_control_factory.tsx

```diff
@@ -59,6 +60,7 @@ export const getTimesliderControlFactory = (): ControlFactory<
       const isAnchored$ = new BehaviorSubject<boolean | undefined>(initialState.isAnchored);
       const isPopoverOpen$ = new BehaviorSubject(false);
       const hasTimeSliceSelection$ = new BehaviorSubject<boolean>(Boolean(timeslice$));
+      const changeRangeByDragging$ = new BehaviorSubject<boolean>(false);
```

Comment:
Hmmm... This feels a tad bit overcomplicated to me :) 

I wonder if we could do the debounce inside the `TimeSliderPopoverContent` instead? Similar to how we do it for the range slider:

https://github.com/elastic/kibana/blob/94f4f669177397e1414704596d60dcae87c03b41/src/plugins/controls/public/controls/data_controls/range_slider/components/range_slider_control.tsx#L54-L61

I did a brief test and it seems to work just as well, and the code feels easier to read IMO.

Reply:
Hi, Hannah! Thank you for the suggestion!

I was wondering if you could tell me how did you manage to use a React hook inside this component? While I was trying to do so, I kept seeing this error.

> React Hook "useCallback" (and "useMemo") is called in function "buildControl" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter. React Hook names must start with the word "use".


In `range_slider_control` EUI component is used, which has an onMouseUp prop whereas in our case we are not directly using an EUI component.
It could be possible if I were to create a wrapping component/element and use a native onMouseUp on that element. I tried a path of adding `onMouseUp` as a prop to `TimeSliderPopoverContent`, but it is a rabbit hole, because we need to define it inside the TimeSliderPopoverContent as well.

Reply:
> React Hook "useCallback" (and "useMemo") is called in function "buildControl" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter. React Hook names must start with the word "use".

This sounds like you tried to add `useMemo` to the `buildControl` method, not the actual React component for the control. You should see that the `buildControl` method returns both an API and a component - so you should have made these changes in the component that is returned (either in the main `Component` or in one of the child components (`TimeSliderPopoverContent`, etc.) 

> I tried a path of adding onMouseUp as a prop to TimeSliderPopoverContent, but it is a rabbit hole, because we need to define it inside the TimeSliderPopoverContent as well.

I think in this case, we would only need the `onMouseUp` in the `TimeSliderPopoverContent`, no? So couldn't we just define it there?